pepsico/projections/maproom.py

```diff
@@ -168,44 +165,307 @@ def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+            ac.read_data("historical", model, variable, region),
+            start_month, end_month,
+            start_year=str(start_year_ref), end_year=str(end_year_ref),
+        ).mean(dim="T"))
+        data = ac.unit_conversion(ac.seasonal_data(
+            ac.read_data(scenario, model, variable, region),
+            start_month, end_month,
+            start_year=str(start_year), end_year=str(end_year),
+        ).mean(dim="T"))
+        data = data - ref
+        if variable in ["hurs", "huss", "pr"]:
+            data = 100. * data / ref
+            data["units"] = "%"
+        return data.rename({"X": "lon", "Y": "lat"})
+
+
+    def map_attributes(variable, data=None):
+        if variable in ["tas", "tasmin", "tasmax"]:
+            map_min = -8
+            map_max = 8
+        elif variable in ["hurs", "huss", "pr"]:
+            map_min = -100
+            map_max = 100
+        else:
+            assert (data is not None)
+            map_amp = max(abs(data.min().values), abs(data.min().values))
+            map_min = -1*map_amp
+            map_max = map_amp
+        return CMAPS["correlation"].rescaled(map_min, map_max), map_min, map_max
 
 
     @APP.callback(
         Output("colorbar", "colorscale"),
         Output("colorbar", "min"),
         Output("colorbar", "max"),
-        Input("location", "pathname"),
+        Input("scenario", "value"),
+        Input("model", "value"),
+        Input("variable", "value"),
+        Input("region", "value"),
+        Input("submit_season","n_clicks"),
+        Input("submit_projy","n_clicks"),
+        Input("submit_refy","n_clicks"),
+        State("start_month", "value"),
+        State("end_month", "value"),
+        State("start_year", "value"),
+        State("end_year", "value"),
+        State("start_year_ref", "value"),
+        State("end_year_ref", "value"),
     )
-    def draw_colorbar(path):
-        scenario = "ssp126"
-        model = "GFDL-ESM4"
-        variable = "tasmin"
-        data = xr.open_zarr(
-            f'/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted'
-            f'/global/monthly_rechunked/{scenario}/{model}/zarr/{variable}'
-        )[variable].isel(T=-1)
-        map_min = data.min().values
-        map_max = data.max().values
-        return CMAPS["rainbow"].to_dash_leaflet(), map_min, map_max
+    def draw_colorbar(
+        scenario,
+        model,
+        variable,
+        region,
+        n_clicks,
+        py_clicks,
+        ry_clicks,
+        start_month,
+        end_month,
+        start_year,
+        end_year,
+        start_year_ref,
+        end_year_ref,
+    ):
+        if variable in ["tas", "tasmin", "tasmax", "hurs", "huss", "pr"]:
+            colorscale, map_min, map_max = map_attributes(variable)
+        else:
+            start_month = ac.strftimeb2int(start_month)
+            end_month = ac.strftimeb2int(end_month)
+            data = seasonal_change(
+            scenario,
```

Comment:
Should be indented another level. Or if this looks right in your editor but not in github, you probably mixed tabs and spaces. Use spaces only.